client/allocrunner/taskrunner/vault_hook.go

```diff
@@ -442,6 +455,12 @@ func (h *vaultHook) deriveVaultTokenJWT() (string, error) {
 		)
 	}
 
+	// If the token cannot be renewed, it doesn't matter if the user set
+	// allow_token_expiration or not, so override the requested behavior
+	if !renewable {
+		h.allowTokenExpiration = false
```

Comment:
```suggestion
		h.allowTokenExpiration = true
```
I think this needs to be true? If the token cannot be renewed then we should allow it to expire.

Reply:
Oops, you're right. That means I'm also missing a test. Fixed and added a test case for it.